### ISSUE TITLE
DRAFT - feat: Improve group and portal sites import strategy

### ISSUE THREAD

**Issue by boubaker** [Meeds-io/MIPs#47]
## Rationale

Currently, when changing the layout of an existing page of when deleting one, the page, we changes the `importMode` of configuration to use one of those values `merge` or `overwrite`. This flag has to remain as is in configurations so that each server startup the `importMode` strategy is reapplied and the whole site is reimported from configuration while we need to import it only once when it changes from sources.

Given this problem, we should design a new type import strategy to fine select the changes to apply on existing installations.

## 1. Functional Requirements

**No functional requirements**

### Configurability 

The new way of defining "import strategy" has to be enabled|disabled by properties.

### Existing Features

The existing `importMode` strategies features has to remain unchanged.

### Feature Flags

No need of feature flags since the new strategy can be enabled/disabled by property.

### Services & processing

To be able to detect changes between startups for each page & navigation, we will have to generate (for imported configurations only) a checksum of each page / navigation node / portal layout.
The checksum will be used only when `importMode` = AUTO in order to know whteher newer changes has to be imported or not.
To make sure to correctly compute the checksum correctly, we can use [XML Digital Signature API](https://docs.oracle.com/javase/9/security/java-xml-digital-signature-api-overview-and-tutorial.htm) (if performant) or just the hashcode of JSON representation of the MOP object.

The pages & navigations imported from sources has to be flagged in database as IMPORTED = true, so that we can distinguish pages imported from sources from those added using UI or space creation ... This will help to delete pages & navigations from DB on startup when deleted from sources automatically and when the selected `importMode` equals to `AUTO`.

**Comment by boubaker** [comment]
Not needed for now since it has been fixed by another less sophisticated solution but sufficient using an Upgrade Plugin in the context of enhancement issue https://github.com/Meeds-io/meeds/issues/1307
